src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -4763,4 +4756,159 @@ public static OptionalDouble getSeverityDuringDecommission()
                OptionalDouble.of(conf.severity_during_decommission) :
                OptionalDouble.empty();
     }
+
+    public static void visit(PropertyVisitor visitor)
+    {
+        visit(visitor, RuntimeException::new);
+    }
+
+    public static void visit(PropertyVisitor visitor, Function<Throwable, ? extends RuntimeException> handler)
+    {
+        try
+        {
+            for (Property property : ofNullable(confValueAccessors).orElse(Collections.emptyMap()).values())
+                visitor.visit(property.getName(), property.getType(), property.getAnnotation(Mutable.class) == null);
+        }
+        catch (Throwable t)
+        {
+            throw handler.apply(t);
+        }
+    }
```

Comment:
I find this API makes the callers less clear, can we delete and replace with something simpler?  such as `Set<String> getProperties()`?